### PR TITLE
nixos-rebuild: add page

### PR DESCRIPTION
**Pull request by aaronduino**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).

This command is central to the NixOS operating system; [official documentation](https://nixos.org/nixos/manual/#sec-changing-config)

### CODE REVIEW

**Comment by aaronduino** [comment]
Turns out I've learned tons about `nixos-rebuild` by writing this tldr page. I think my modifications are complete (apart from acting on maintainer feedback). I didn't intend to add anything after the initial submission.

File: pages/common/nixos-rebuild.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# nixos-rebuild
+
+> Reconfigure a NixOS machine.
+
+- Build and switch to the new configuration, making it the boot default:
```
Comment (mebeim) [inline pages/common/nixos-rebuild.md]: I see this example has basically the same description as the 5th one. What's the difference? I think one of them (if not both) should be reworded to be clearer.

File: pages/common/nixos-rebuild.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+- Build and switch to the new configuration, making it the boot default and installing updates:
+
+`sudo nixos-rebuild switch --upgrade`
+
+- Rollback changes to the configuration, switching to the previous generation:
+
+`sudo nixos-rebuild switch --rollback`
+
+- Build the new configuration and make it the boot default:
```
Comment (mebeim) [inline pages/common/nixos-rebuild.md]: See above.

**Comment by aaronduino** [comment]
@mebeim the way it works is that nixos-rebuild does a combination of switching the OS to a new configuration while it is running and adding a boot entry,

Name | Switch | Boot
---|---|---
Switch| x| x
Boot||x
Test|x|

**Comment by aaronduino** [comment]
I'm moving this to the Linux folder since it is only a available on the NixOS distro of Linux.

File: pages/common/nixos-rebuild.md (hunk 2 above)
Comment (aaronduino) [reply pages/common/nixos-rebuild.md]: Does the change I made fix this issue? https://github.com/tldr-pages/tldr/pull/2809/commits/926f8a8c4889d5859cf8c90b5bdfcc4143a167f0

File: pages/linux/nixos-rebuild.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# nixos-rebuild
+
+> Reconfigure a NixOS machine.
```
Comment (sbrl) [inline pages/linux/nixos-rebuild.md]: Perhaps we could add an additional line below this one that links to the documentation with the new homepage syntax we drafted up in #2649?

```
> Documentation: https://nixos.org/nixos/manual/#sec-changing-config
```

Comment (aaronduino) [reply pages/linux/nixos-rebuild.md]: Okay, I just committed the above change

**Comment by aaronduino** [comment]
Is that true? Couldn't adding a period to a URL be problematic?

File: pages/linux/nixos-rebuild.md (hunk 3 above)
Comment (sbrl) [reply pages/linux/nixos-rebuild.md]: Cool! I've just realised that I got the syntax incorrect in my above comment. I've made quick commit to fix it, if you don't mind?

**Comment by sbrl** [comment]
Oops, @aaronduino! See my comment on the original thread.

File: pages/linux/nixos-rebuild.md (hunk 3 above)
Comment (aaronduino) [reply pages/linux/nixos-rebuild.md]: No problem at all. Thanks!

**Pull request merged**